src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Carnivore) {
+      return;
+    }
+
+    if (!target.hidden && target instanceof Herbivore) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        const deadAnimalIndex = Animal.alive.indexOf(target);
+
+        if (deadAnimalIndex !== -1) {
+          Animal.alive.splice(deadAnimalIndex, 1);
+        }
```

Comment:
Instead of using indexOf and splice within a loop, which can be inefficient, consider using the filter method to create a new array without the dead animal. This approach avoids modifying the array during iteration and can improve performance.